src/test/kotlin/org/rust/cargo/project/model/CargoPackagesTest.kt

```diff
@@ -90,6 +90,47 @@ class CargoPackagesTest : RsWithToolchainTestBase() {
         workspace.checkPackage("windows_package", shouldExist = true)
     }
 
+    @MinRustcVersion("1.64.0")
+    fun `test target specific dependencies with multiple build targets`() {
+        buildProject {
+            dir(".cargo") {
+                toml("config.toml", """
+                    [build]
+                    target = ["x86_64-pc-windows-msvc", "i686-pc-windows-msvc"]
+                """)
+            }
+            toml("Cargo.toml", """
+                [package]
+                name = "sandbox"
+                version = "0.1.0"
+                authors = []
+
+                [dependencies]
+                common_package = { path = "common_package" }
+
+                [target.'cfg(unix)'.dependencies]
+                unix_package = { path = "unix_package" }
+
+                [target.'cfg(windows)'.dependencies]
+                windows_package = { path = "windows_package" }
+            """)
+
+            dir("src") {
+                rust("lib.rs", "")
+            }
+
+            dependencyPackage("common_package")
+            dependencyPackage("unix_package")
+            dependencyPackage("windows_package")
+        }
+
+        val workspace = project.cargoProjects.singleProject().workspaceOrFail()
+
+        workspace.checkPackage("common_package", shouldExist = true)
+        workspace.checkPackage("unix_package", shouldExist = false)
+        workspace.checkPackage("windows_package", shouldExist = true)
```

Comment:
This test don't actually test if IntelliJ Rust considered all specified targets. For expample, it will pass if the implementation takes into account only the first target (`x86_64-pc-windows-msvc`) but ignores the second one.

Reply:
Good point. I'll change the test dependencies to demonstrate processing both targets.

There is already a lower level test for this in CargoConfigTest, which checks that multiple targets are correctly picked up in the config. But this test will also confirm that Rust is selecting the dependencies across multiple targets correctly.